src/Symfony/Component/Serializer/Normalizer/BackedEnumNormalizer.php

```diff
@@ -45,6 +50,18 @@ public function denormalize(mixed $data, string $type, string $format = null, ar
             throw new InvalidArgumentException('The data must belong to a backed enumeration.');
         }
 
+        if (true === ($context[self::ALLOW_INVALID_VALUES] ?? false)) {
+            if (null === $data) { // "tryFrom()" does not accept null
+                return null;
+            }
+
+            try {
+                return $type::tryFrom($data);
+            } catch (\TypeError) {
```

Comment:
Try catch is costly and should be avoided on the hot path. Do we really need to allow other types than string and int? We don't catch the `TypeError` when calling `from()`.

If that's really necessary, we should test that the value is a string or an int before calling `tryFrom()` instead.

Reply:
> If that's really necessary

I'd say if you don't control the data which is serialized, it could be anything (I presently have a use case where this whole thing would be useful)

> Try catch is costly and should be avoided on the hot path.

:exploding_head: just learnt something!

The problem here is there is no way to know if the enum is a "string backed enum" or "int backed enum". So if you have a string, but your enum is a "int backed enum" there will be an error. This case should be very rare I guess. I've added to exclude empty strings, which is actually one of the cases that made me create this

Reply:
@dunglas actually I made a mistake here: I should not exclude "empty string" because it is [a valid value for a string backed enum](https://3v4l.org/dXM9H#v8.2.1)

then it would be very useful to keep the `try/catch` stuff because otherwise I could not return null if the enum is "int backed" and the value is an empty string, which is exactly one of the use cases for which I created this PR. (that's why the CI fails)

Not sure we are in a "hot path" here, because it will only hit the try/catch for deserialization in enum properties if this specific context is applied